framework/src/tinyalsa/tinyalsa.c

```diff
@@ -311,6 +308,16 @@ int pcm_set_config(struct pcm *pcm, const struct pcm_config *config)
 		pcm->config = *config;
 	}
 
+	if (ioctl(pcm->fd, AUDIOIOC_GETBUFFERINFO, (unsigned long)&buf_info) < 0) {
```

Comment:
We are calculating period size based on sample rate. What happens if user changes sample rate or other parameters later?

Reply:
set_config should called before pcm_start.
It is violation by user.

Reply:
How about

Reply:
yes. so alloc should be moved to other function, pcm_start maybe.